src/taskEvents/logCard.js

```diff
@@ -0,0 +1,170 @@
+      alt: 'dropdown icon',
+    },
+  });
+
+  summaryContainer.appendChild(tasklog);
+  summary.appendChild(summaryContainer);
+  summary.appendChild(icon);
+
+  return summary;
+}
+
+function createDetailsSection(title, purpose, category, level) {
+  const detailsContainer = createElement({
+    type: 'div',
+    attributes: { class: 'details-div-container' },
+  });
+  const details = createElement({
+    type: 'div',
+    attributes: { class: 'details-div' },
+  });
+
+  const taskTitleDiv = createElement({
+    type: 'div',
+    attributes: { class: 'task-title-div' },
+  });
+  const taskTitleLabel = createElement({
+    type: 'span',
+    attributes: { class: 'task-title-label' },
+    innerText: 'Title: ',
+  });
+  const titleDetail = createElement({
+    type: 'span',
+    attributes: { class: 'task-title-detail' },
+    innerText: title,
+  });
+
+  taskTitleDiv.appendChild(taskTitleLabel);
+  taskTitleDiv.appendChild(titleDetail);
+
+  details.appendChild(taskTitleDiv);
+
+  const taskPurposeDiv = createElement({
+    type: 'div',
+    attributes: { class: 'task-purpose-div' },
+  });
+  const taskPurposeLabel = createElement({
+    type: 'span',
+    attributes: { class: 'task-purpose-label' },
+    innerText: 'Purpose: ',
+  });
+  const taskPurposeDetail = createElement({
+    type: 'span',
+    attributes: { class: 'task-purpose-detail' },
+    innerText: purpose,
+  });
+
+  taskPurposeDiv.appendChild(taskPurposeLabel);
+  taskPurposeDiv.appendChild(taskPurposeDetail);
+
+  details.appendChild(taskPurposeDiv);
+
+  if (category) {
+    const taskCategoryDiv = createElement({
+      type: 'div',
+      attributes: { class: 'task-cateogory-div' },
+    });
+    const taskCategoryLabel = createElement({
+      type: 'span',
+      attributes: { class: 'task-category-label' },
+      innerText: 'category: ',
+    });
+    const taskCategoryDetail = createElement({
+      type: 'span',
+      attributes: { class: 'task-category-detail' },
+      innerText: category,
+    });
+
+    taskCategoryDiv.appendChild(taskCategoryLabel);
+    taskCategoryDiv.appendChild(taskCategoryDetail);
+    details.appendChild(taskCategoryDiv);
+  }
+
+  if (level) {
+    const taskLevelDiv = createElement({
+      type: 'div',
+      attributes: { class: 'task-level-div' },
+    });
+    const taskLevelLabel = createElement({
+      type: 'span',
+      attributes: { class: 'task-level-label' },
+      innerText: 'Level: ',
+    });
+    const taskLevelDetail = createElement({
+      type: 'span',
+      attributes: { class: 'task-level-detail' },
+      innerText: level,
+    });
+
+    taskLevelDiv.appendChild(taskLevelLabel);
+    taskLevelDiv.appendChild(taskLevelDetail);
+    details.appendChild(taskLevelDiv);
+  }
+
+  detailsContainer.appendChild(details);
+
+  return detailsContainer;
+}
+
+function createEventCard(
+  container,
+  title,
+  eventdescription,
+  purpose,
+  username,
+  category,
+  level,
+  isAllTasks,
+  createModal,
```

Comment:
It would be good if we can wrap these arguments as an object so that they could be passed in any order